release/nightly_tests/dataset/iter_batches_benchmark.py

```diff
@@ -0,0 +1,99 @@
+import sys
+from typing import Optional
+
+import ray
+from ray.data.dataset import Dataset
+
+from benchmark import Benchmark
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+
+def iter_batches(
+    ds: Dataset,
+    batch_size: Optional[int] = None,
+    batch_format: Literal["default", "pandas", "pyarrow", "numpy"] = "default",
+    local_shuffle_buffer_size: Optional[int] = None,
+    use_default_params: bool = False,
+) -> Dataset:
+    num_batches = 0
+    if use_default_params:
+        for batch in ds.iter_batches():
+            num_batches += 1
+    else:
+        for batch in ds.iter_batches(
+            batch_format=batch_format,
+            batch_size=batch_size,
+            local_shuffle_buffer_size=local_shuffle_buffer_size,
+        ):
+            num_batches += 1
+    print("iter_batches done, num_rows:", ds.count(), "num_batches:", num_batches)
+    return ds
+
+
+def run_iter_batches_benchmark(benchmark: Benchmark):
```

Comment:
This benchmark is geared towards tabular data, we should also have a benchmark on CV data, e.g. on one of the imagery datasets we're hosting in S3, either synthetic or something like ImageNet. Batching over a 17 scalar-column dataset might have quite different performance characteristics than batching over a single tensor column.

Given that CV is a top priority for us, it's probably higher priority to have a CV-oriented benchmark than a tabular-oriented benchmark. cc @c21

Reply:
Yeah good callout. Agree this can be in a separate PR for followup. E.g. `s3://air-example-data-2/1G-image-data-synthetic-raw` has ImageNet data, which is used by batch prediction benchmark.

Reply:
Thanks for bringing up, it's a good area to test out!